### ISSUE TITLE
mkv with picture/attachments

### ISSUE THREAD

**Issue by zbijelic** [rigaya/NVEnc#221]
Hello,

A .mkv with a picture poster.jpg or a background.jpg is considered as video and doesn't encode.

avsw: first key frame had timestamp AV_NOPTS_VALUE, this might lead to avsync error

I have removed the attachments with mkvtools and it's working, maybe you can add --copy-attachments, I think it's -disposition in ffmpeg

Great project ! Bravo !

**Comment by rigaya** [comment]
I have updated to [NVEnc 5.01](https://github.com/rigaya/NVEnc/releases/tag/5.01), which will check if the track should be taken as attachments and should not to be considered as video, so you shall not get the error you have pointed out anymore.

> avsw: first key frame had timestamp AV_NOPTS_VALUE, this might lead to avsync error

I've also tried adding [--attachment-copy](https://github.com/rigaya/NVEnc/blob/master/NVEncC_Options.en.md#--attachment-copy-intint), and it works when the output is mp4 format. For mkv format, it's little different, the attachment is taken as mjpeg stream like cover art. (not the same as mkvtools?)
